subsys/net/ip/net_pkt.c

```diff
@@ -1704,9 +1718,21 @@ int net_pkt_get_src_addr(struct net_pkt *pkt, struct sockaddr *addr,
 		proto = NET_IPV4_HDR(pkt)->proto;
 
 		if (IS_ENABLED(CONFIG_NET_TCP) && proto == IPPROTO_TCP) {
-			addr4->sin_port = net_pkt_tcp_data(pkt)->src_port;
+			struct net_tcp_hdr hdr;
```

Comment:
We have lot of duplicate code now. Could this be refactored a bit so that we would get the TCP and UDP header only once? Earlier we had only one function in each branch but now this PR adds more code so some re-factoring could be done.